trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/library/FunctionModelHelper.java

```diff
@@ -0,0 +1,358 @@
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.result.ExpressionResultBuilder;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.result.LRValueFactory;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.result.LocalLValue;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.result.RValue;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.result.Result;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.util.SFO;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.util.SFO.AUXVAR;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.interfaces.handler.ITypeHandler;
+import de.uni_freiburg.informatik.ultimate.core.lib.models.annotation.Check;
+import de.uni_freiburg.informatik.ultimate.core.lib.models.annotation.CheckMessageProvider;
+import de.uni_freiburg.informatik.ultimate.core.lib.models.annotation.Overapprox;
+import de.uni_freiburg.informatik.ultimate.core.model.models.ILocation;
+import de.uni_freiburg.informatik.ultimate.core.model.models.annotation.Spec;
+
+/**
+ * Helper class for implementations of {@link ILibraryModel}
+ */
+public class FunctionModelHelper {
+	private final ExpressionTranslation mExpressionTranslation;
+	private final MemoryHandler mMemoryHandler;
+	private final AuxVarInfoBuilder mAuxVarInfoBuilder;
+	private final TypeSizes mTypeSizes;
+	private final ITypeHandler mTypeHandler;
+	private final boolean mCheckMemoryLeakInMain;
+	private final boolean mSvcompMemtrackCompatibilityMode;
+
+	public FunctionModelHelper(final AuxVarInfoBuilder auxVarInfoBuilder,
+			final ExpressionTranslation expressionTranslation, final MemoryHandler memoryHandler,
+			final TypeSizes typeSizes, final ITypeHandler typeHandler, final boolean checkMemoryLeakInMain,
+			final boolean svcompMemtrackCompatibilityMode) {
+		mExpressionTranslation = expressionTranslation;
+		mMemoryHandler = memoryHandler;
+		mAuxVarInfoBuilder = auxVarInfoBuilder;
+		mTypeSizes = typeSizes;
+		mTypeHandler = typeHandler;
+		mCheckMemoryLeakInMain = checkMemoryLeakInMain;
+		mSvcompMemtrackCompatibilityMode = svcompMemtrackCompatibilityMode;
+	}
+
+	/**
+	 * Check if the expected number of arguments matches the actual size of {@code arguments} and throws an exception
+	 * otherwise.
+	 */
+	public void checkArguments(final ILocation loc, final int expectedArgs, final String name,
+			final IASTInitializerClause[] arguments) {
+		if (arguments.length != expectedArgs) {
+			throw new IncorrectSyntaxException(loc, name + " is expected to have " + expectedArgs
+					+ " arguments, but was called with " + arguments.length);
+		}
+	}
+
+	/**
+	 * Handle a function call by dispatching all arguments and then calling a function with no arguments that has the
+	 * name of the function and is marked with the {@link Overapprox} annotation. Additionally it is assumed that the
```

Comment:
This documentation is no longer correct, as far as I can determine. Could you please update it?